dropletutils-read-10x-counts.R

```diff
@@ -34,19 +67,50 @@ option_list = list(
 opt <- wsc_parse_args(option_list, mandatory = c('samples', 'output_object_file'))
 
 # Now we're hapy with the arguments, load DropletUtils and do the work
-
 suppressPackageStartupMessages(require(DropletUtils))
 
 # Parse out multiple directories from the samples string
-
 samples <- wsc_split_string(opt$samples)
 
 # Derive the SingleCellExperiment
+single_cell_experiment <- read10xCounts( samples = samples, col.names = opt$col_names )
+
+# metadata file logic 
+if(!is.na(opt$metadata_files)){
+  metadata_files = wsc_split_string(opt$metadata_files)
+  # check cell id column is provided
+  if(is.na(opt$cell_id_column)){
+    stop("Cell id column name for SDRF files is not provided. ")
+  }
+  metadata = lapply(metadata_files, function(x) read.csv(x, sep="\t"))
+  metadata = do.call(rbind, metadata)
```

Comment:
I believe rbind() assumes the same number and identity of the columns. If I were you I would find the columns common to all metadata files and use those when combining the data frames, like:

```
common_names <- Reduce(intersect, lapply(metadata, colnames))
metadata = do.call(rbind, lapply(metadata, function(x) x[,comon_names]))
```